vsc_pdf_template_transformer/lib/models/tpl_page.dart

```diff
@@ -24,8 +34,16 @@ class TplPage {
   Map<String, dynamic> toJson() => _$TplPageToJson(this);
 
   pw.Page toPdf(Map<String, dynamic> data) {
-    return pw.Page(build: (pw.Context context) {
-      return child?.buildWidget(data) ?? pw.SizedBox();
-    });
+    return pw.Page(
+        pageTheme: pageTheme?.toPdf(data),
+        theme: theme?.toPdf(data),
+        pageFormat: evaluatePageFormat(pageFormat, data),
```

Comment:
Although there are standard sizes for PdfPageFormat, you can also specify your own page format, most importantly with  margins. I think we need a TplPdfPageFormat class with the individual PdfPageFormat values, plus a `format` attribute which evaluates to a string if it exists. The string would be a3, a4, letter, etc.